iocage_cli/export.py

```diff
@@ -21,16 +22,75 @@
 # STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING
 # IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
 # POSSIBILITY OF SUCH DAMAGE.
-"""export module for the cli."""
+"""Export a jail from the CLI."""
 import click
+import os.path
 
-import iocage_lib.iocage as ioc
+import iocage.errors
+import iocage.Filter
+import iocage.Jail
+import iocage.Jails
+import iocage.Logger
+import iocage.Releases
+import iocage.Resource
+
+from .shared.click import IocageClickContext
 
 __rootcmd__ = True
 
 
-@click.command(name="export", help="Exports a specified jail.")
+@click.command(name="export", help="Export a jail to a backup archive")
+@click.pass_context
 @click.argument("jail", required=True)
-def cli(jail):
-    """Make a recursive snapshot of the jail and export to a file."""
-    ioc.IOCage(jail=jail).export()
+@click.argument("destination", required=True)
+@click.option(
+    "-s", "--standalone",
+    default=False,
+    is_flag=True,
+    help="Exports the jails root dataset independently"
+)
+# @click.option(
+#     "-r", "--recursive",
+#     default=False,
+#     is_flag=True,
+#     help="Include ZFS snapshots. Implies --standalone"
+# )
+def cli(
+    ctx: IocageClickContext,
+    jail: str,
+    destination: str,
+    standalone: bool
+) -> None:
+    """
+    Backup a jail.
+
+    The selected jail will be exported to a gzip compressed tar archive stored
+    as the destination path.
+    """
+    logger = ctx.parent.logger
+    zfs: iocage.ZFS.ZFS = ctx.parent.zfs
+    host: iocage.Host.HostGenerator = ctx.parent.host
+    print_events = ctx.parent.print_events
+
+    # Recursive exports cannot be imported at the current time
+    recursive = False
+
+    ioc_jail = iocage.Jail.JailGenerator(
+        jail,
+        logger=logger,
+        zfs=zfs,
+        host=host
+    )
+
+    if os.path.isfile(destination) is True:
```

Comment:
Should this be os.path.exists ? In case someone try using a directory or something that is not a regular file.

Reply:
Yes, that's indeed undefined here. When it is a directory, would you expect some automatically generated file name or to see the same error message?

Reply:
In my opinion the same error message would be OK